### PR TITLE
Add two new options for merge checks

### PR DESCRIPTION
**Pull request by mmonsenhexagon**
Add two new options for merge checks: Toggle result caching, and Include Details in summary (merge button tooltip)

I coded up these new options because we have need for them. We have been using an older version of the External Hooks plugin that had this behavior, but after upgrading we discovered that the new version had changed behavior.

First, we need our merge check results to be run each time the page is refreshed, rather than cached. This is because one of our hooks checks to see if the branch being merged has already been merged into a different branch. If it hasn't been merged yet, we return information about where the branch needs to be merged before this pull request can proceed. If this result is cached, then even if the user goes and merges the branch where it needs to go, the pull request will continue to show the error (since the merge doesn't change the current branch or the target branch.)

Second, we prefer to see the details of the check failure displayed in the tooltip of the Merge button, rather than as a comment to the pull request. This is especially important given the issue described in the paragraph above. If we didn't cache the results, we would get multiple comments logged to the pull request.

In summary, our normal workflow is to always have the merge check run, and to display the details of a failure in the tooltip of the Merge button (i.e. return the details in the summary portion of the RepositoryHookResult.)

If you would like any changes to be made, or code cleanup, I'll be happy to work on them. Thanks for considering my pull request.

### CODE REVIEW

File: src/main/resources/static/external-merge-check-hook.soy [hunk 1]
```diff
@@ -61,4 +61,26 @@
...
+            [
+                'id': 'cache_pr_results',
+                'labelText': 'Cache results of pull request check',
+                'isChecked': $config ? $config['cache_pr_results'] : true
+            ]
+        ]/}
+    {/call}
+	
+	{call aui.form.checkboxField}
+        {param legendContent: 'Include Details In Summary' /}
```
Comment (seletskiy) [inline src/main/resources/static/external-merge-check-hook.soy]: Please rename it to "Include Details in Tooltip".

File: src/main/java/com/ngs/stash/externalhooks/hook/ExternalMergeCheckHook.java [hunk 2]
```diff
@@ -255,6 +267,7 @@ public RepositoryHookResult preUpdate(PreRepositoryHookContext context, PullRequ
         final String summaryMsg = wasAccepted ? "External hooks check was successful." : "External hooks check failed.";
 
         check.setWasAccepted(wasAccepted);
+        check.setLastExceptionSummary(String.format("%s\n%s", summaryMsg, detailMsg));
```
Comment (seletskiy) [inline src/main/java/com/ngs/stash/externalhooks/hook/ExternalMergeCheckHook.java]: Result of this set will be overridden by next line, maybe there you need to set it conditionally depending on value of `tooltip_results`?

File: src/main/resources/static/external-merge-check-hook.soy [hunk 3]
```diff
@@ -61,4 +61,26 @@
...
+        {param fields: [
+            [
+                'id': 'tooltip_results',
```
Comment (seletskiy) [inline src/main/resources/static/external-merge-check-hook.soy]: Can you rename it to `detailed_tooltip`?

File: src/main/resources/static/external-merge-check-hook.soy [hunk 4]
```diff
@@ -61,4 +61,26 @@
             ]
         ]/}
     {/call}
+	
+	{call aui.form.checkboxField}
+        {param legendContent: 'Cache Check Results' /}
+        {param fields: [
+            [
+                'id': 'cache_pr_results',
+                'labelText': 'Cache results of pull request check',
```
Comment (seletskiy) [inline src/main/resources/static/external-merge-check-hook.soy]: Can you please also add `descriptionText` with something like:

```
Without this option specified executable will be executed whenever user will open pull request page.
```

**Comment by seletskiy** [comment]
Hey @mmonsenhexagon!

Thanks for valuable contribution.

There are little nitpicks that I would like to get fixed before merge.

**Comment by seletskiy** [comment]
Also, can you please split commit message into short summary and more detailed body, since long summary (>69 chars) doesn't fit in GitHub interface and is generally considered not good practice.

File: src/main/java/com/ngs/stash/externalhooks/hook/ExternalMergeCheckHook.java (hunk 2 above)
Comment (mmonsenhexagon) [reply src/main/java/com/ngs/stash/externalhooks/hook/ExternalMergeCheckHook.java]: I think this was mistakenly left in during my development. I had meant to keep the line that follows.

**Comment by mmonsenhexagon** [comment]
Hello,

I have made the changes you requested. I amended the commit to clean up the commit message and did a force-push to replace my previous commit. I hope that the wording I used on the descriptions is OK for you. Thanks again for considering my pull request.

**Comment by mmonsenhexagon** [comment]
Also did another update to fix some formatting issues.

File: src/main/resources/static/external-merge-check-hook.soy (hunk 1 above)
Comment (mmonsenhexagon) [reply src/main/resources/static/external-merge-check-hook.soy]: Renamed.

File: src/main/resources/static/external-merge-check-hook.soy (hunk 3 above)
Comment (mmonsenhexagon) [reply src/main/resources/static/external-merge-check-hook.soy]: Renamed.

File: src/main/resources/static/external-merge-check-hook.soy (hunk 4 above)
Comment (mmonsenhexagon) [reply src/main/resources/static/external-merge-check-hook.soy]: Added.

**Comment by seletskiy** [comment]
@mmonsenhexagon: thank you very much!

I will do thorough testing of new changes on my test setup soon and then will release new version on marketplace.
